classes/class-wc-connect-logger.php

```diff
@@ -2,16 +2,21 @@
 
 if ( ! class_exists( 'WC_Connect_Logger' ) ) {
 
+	define( 'WOOCOMMERCE_CONNECT_DEBUG_LOGGING_ENABLED_OPTION', 'wcc_debug_logging_enabled' );
+
 	class WC_Connect_Logger {
 
 		/**
 		 * @var WC_Logger
 		 */
 		private $logger;
 
+		private $is_logging_enabled = false;
+
 		public function __construct( WC_Logger $logger ) {
 
 			$this->logger = $logger;
+			$this->is_logging_enabled = ( 1 == get_option( WOOCOMMERCE_CONNECT_DEBUG_LOGGING_ENABLED_OPTION, 0 ) );
```

Comment:
OK, I tested this and remembered why I needed this: update_option with true results in a string 1 being returned by get_option.

Similarly, update_option with false results in an empty string being returned by get_option

Here's an article that describes the stringification behavior a bit - https://www.storelocatorplus.com/wordpress-boolean-options-and-javascript-handle-with-care/

Net net - I'm going to change that get_option to a strict === comparison for string '1' and make the default a string '0'